clintrials/dosefinding/efftox.py

```
@@ -344,48 +344,78 @@ def __call__(self, prob_eff, prob_tox):
             return 1 - r_to_the_p ** (1.0 / self.p)
         else:
+            if np.isscalar(x) and np.isscalar(y):
+                return np.nan
             response = np.zeros_like(x, dtype=float)
-            response *= np.nan
+            response[:] = np.nan
             return response
 
-    def solve(self, prob_eff=None, prob_tox=None, delta=0):
+    def solve(
+        self, delta, *, prob_eff=None, prob_tox=None, bounds=(0, 1), tol=1e-6, maxiter=100
```

Ruff (0.15.18)

[warning] Unused method argument: `tol`

(ARG002)